mllib/src/main/scala/org/apache/spark/ml/regression/IsotonicRegression.scala

```diff
@@ -84,7 +84,7 @@ private[regression] trait IsotonicRegressionBase extends Params with HasFeatures
       val extract = udf { v: Vector => v(idx) }
       extract(col($(featuresCol)))
     } else {
-      col($(featuresCol))
+      col($(featuresCol)).cast(DoubleType)
```

Comment:
Precision loss may happen here. But I think we can not avoid it now, for we use `mllib.IsotonicRegression` and the input dataframe is of type `input: RDD[(Double, Double, Double)])`